ui-tests/src/test/java/org/openmrs/reference/page/PatientVisitsDashboardPage.java

```diff
@@ -44,9 +49,29 @@ public PatientVisitsDashboardPage(Page parent, WebElement waitForStaleness) {
     public String getPageUrl() {
         return "coreapps/patientdashboard/patientDashboard.page";
     }
+    
+    public PatientCaptureVitalsPage goToPatientCaptureVitalsPage() {
+        waiter.until(ExpectedConditions.presenceOfElementLocated(CAPTURE_VITALS));
+        clickOn(CAPTURE_VITALS);
+        return new PatientCaptureVitalsPage(this);
+    }
+    
+    public EditVitalsPage goToEditVitalsPage() {
+        waiter.until(ExpectedConditions.presenceOfElementLocated(EDIT_VITALS_ICON));
+        clickOn(EDIT_VITALS_ICON);
+        return new EditVitalsPage(this);
+    }
 
-    public void goToCaptureVitals() {
-        findElement(CAPTURE_VITALS).click();
+    public void clickOnViewVitalsIcon() {
+        waiter.until(ExpectedConditions.presenceOfElementLocated(VIEW_VITALS_ICON));
+        clickOn(VIEW_VITALS_ICON);
+    }
+
+    public void deleteVitals() {
+        waiter.until(ExpectedConditions.presenceOfElementLocated(DELETE_VITALS_ICON));
+        clickOn(DELETE_VITALS_ICON);
+        waitForElement(COMFIRM_DELETE_BUTTON);
```

Comment:
What happens when you dont have this line 73

Reply:
So ideally which error does it return when it takes long to be clicked

Reply:
When a delete action is invoked on the delete button, its a good design to have it throw a confirm dialogue for the user to confirm the action. There is a milliseconds taken for the confirm dialogue to be rendered on the UI and so requires to invoke waitForElement() function for the next action to be effective.